src/components/Recipes/Recipes.tsx

```diff
@@ -10,21 +10,29 @@ import './Recipes.css';
 const Recipes: React.FC = () => {
   const foodprintCtx = useContext(FoodprintContext);
   const setRecipes: (title: string, id: number, image: string) => void = foodprintCtx.recipes.setRecipes;
-  const { items } = foodprintCtx.recipes;
-  const ingredientList = foodprintCtx.ingredients.items;
+  const recipeList = foodprintCtx.recipes.items;
+  const { items, queryString, setQueryString } = foodprintCtx.ingredients;
+  const ingredientList = items;
   const timeout = useRef();
-  const [isLoading, setIsLoading] = useState(false);
+  const [ isLoading, setIsLoading ] = useState(false);
+
+  const debouncer = useCallback<any>(() => {debounce( 1400, setIsLoading, setQueryString, ingredientList, timeout );}, [ingredientList]);
 
-  const debouncer = useCallback<any>(() => {debounce(fetchData, 1400, setIsLoading, setRecipes, ingredientList, timeout);}, [ingredientList]);
 
   useEffect(() => {
-    if (ingredientList.length) {
-      debouncer();
+    if (ingredientList.length > 0) {
+      debounce( 1400, setIsLoading, setQueryString, ingredientList, timeout )
     }
   }, [ingredientList]);
+  useEffect(() => {
+    if (queryString) {
```

Comment:
We might need a mounted variable here, to prevent multiple requests

Reply:
refer to 13:50 of this video https://www.youtube.com/watch?v=QQYeipc_cik

video in general is helpful but 13:50 to what im referring to here